### PR TITLE
chore(ux): optimize progress bar

### PR DESCRIPTION
**Pull request by qweeah**
This PR
1) Adds defensive code to avoid showing negative speed in progress bar.
2) Calculates the download speed based on the size of transferred data in the past last seconds.

### CODE REVIEW

File: cmd/oras/internal/display/status/progress/speed.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package progress
+
+type speedWindow struct {
+	point []int64
```
Comment (shizhMSFT) [inline cmd/oras/internal/display/status/progress/speed.go:19]: According to your algorithm for accuracy, a point should be a pair of `(time, offset)`. Then the average time is $$\text{Average Time} = \frac{\text{head}.offset - \text{tail}.offset}{\text{head}.time - \text{tail}.time}$$.

Comment (shizhMSFT) [reply cmd/oras/internal/display/status/progress/speed.go:19]: We can discuss the algorithm offline if needed

**Review by shizhMSFT — CHANGES_REQUESTED** [review]

File: cmd/oras/internal/display/status/progress/speed.go (hunk 1 above)
Comment (qweeah) [reply cmd/oras/internal/display/status/progress/speed.go:19]: yes you are right, will change accordingly.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**